packages/aws-cdk-lib/aws-codepipeline/lib/pipeline.ts

```diff
@@ -495,6 +521,14 @@ export class Pipeline extends PipelineBase {
     }
     this.pipelineType = props.pipelineType ?? PipelineType.V1;
 
+    if (
+      props.executionMode
+      && [ExecutionMode.QUEUED, ExecutionMode.PARALLEL].includes(props.executionMode)
+      && this.pipelineType !== PipelineType.V2
+    ) {
+      throw new Error(`${props.executionMode} execution mode can only be used with V2 pipelines, \`PipelineType.V2\` must be specified for \`pipelineType\``);
+    }
+
```

Comment:
I have confirmed that the error occurs when QUEUED or PARALLEL mode is specified in the AWS console.

```
QUEUED or PARALLEL mode can only be used with V2 pipelines
```